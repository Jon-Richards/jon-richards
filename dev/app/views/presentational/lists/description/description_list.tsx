import * as React from 'react';
import { List } from '../interfaces/list';
import { DescriptionListItem } from '../interfaces/description_list_item';

export function DescriptionList(props: DescriptionListProps): JSX.Element {
  const items = mapItems(props.items);
  return (
    <dl title={props.title}>
      {items}
    </dl>
  )
}

type DescriptionListProps = List<DescriptionListItem>;

function mapItems(items: DescriptionListItem[]): JSX.Element {
  const result = items.map(item => {
    if (item.type === 'TERM') {
      return createDescriptionTerm(item);
    } else {
      return createDescriptionListItem(item);
    }
  });
  return <React.Fragment>{result}</React.Fragment>
}

function createDescriptionTerm(item: DescriptionListItem): JSX.Element {
  return <dt key={item.id}>{item.content}</dt>
}

function createDescriptionListItem(item: DescriptionListItem): JSX.Element {
  return <dd key={item.id}>{item.content}</dd>
}
