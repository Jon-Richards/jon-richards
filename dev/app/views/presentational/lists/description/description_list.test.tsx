import * as React from 'react';
import { DescriptionList } from './description_list';
import { render } from 'enzyme';

describe('The OrderedList component.', () => {
  it('Should match the snapshot.', () => {
    const wrapper = render(
      <DescriptionList
        title="An Ordered List."
        items={[
          { id: 'one', type: 'TERM', content: 'A Term' },
          { id: 'two', type: 'DESCRIPTION', content: 'Description one' },
          { id: 'two', type: 'DESCRIPTION', content: 'Description two' }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
