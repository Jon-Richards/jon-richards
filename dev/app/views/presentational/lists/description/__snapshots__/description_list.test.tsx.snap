// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The OrderedList component. Should match the snapshot. 1`] = `
<dl
  title="An Ordered List."
>
  <dt>
    A Term
  </dt>
  <dd>
    Description one
  </dd>
  <dd>
    Description two
  </dd>
</dl>
`;
